src/Compiler/LatteToPhpCompiler.php

```diff
@@ -37,7 +43,14 @@ public function compileFile(Template $template, string $context = ''): string
         $phpContent = $this->postprocessor->postProcess($phpContent, $template);
         $templateDir = pathinfo($templatePath, PATHINFO_DIRNAME);
         $templateFileName = pathinfo($templatePath, PATHINFO_BASENAME);
-        $contextHash = md5($template->getSignatureHash() . $context);
+        $contextHash = md5(
+            $template->getSignatureHash() .
+            $context .
+            $this->cacheKey .
+            Engine::VERSION_ID .
+            PHP_VERSION_ID .
+            (class_exists(InstalledVersions::class) ? json_encode(InstalledVersions::getAllRawData()) : '')
```

Comment:
Is this Installed Version from app or phpstan?

Reply:
This is class defined by composer autoloader. It gibes you info about what packages composer installed.

Reply:
I know, but I have phpstan installed as standalone app and not in tested application so if  I run it I’m not sure what packages will be loaded

Reply:
Inam not sure too. But i would guess global version. Because it is what is used to load phpstan. But i think it is what we want because it is also what we use to compile.

Reply:
Ah I thought you want to know nette packages' versions

Reply:
I’ll try it which composer it will use

Reply:
Basically I want to know if anythong that can affect how compiled version looks like changes.

Reply:
Well it is affected by NodeVisitors of this extension and also by nette app itself - for example last change in nette/forms

Reply:
Yes. I think phpstan don’t need autoloading anymore. But before I just add paths to bootstrap.php to phpstan config.

Reply:
Well it needs to autoload Latte classes somehow to use to compile templates.

Reply:
they are included in this package https://github.com/efabrica-team/phpstan-latte/blob/main/composer.json#L9

Reply:
So if you use global version you use latte version independent from your app to compile templates. This code woukd get versions of latte used to compile in global install - which is what we want.

Reply:
I checked this now and I have both (phpstan global app and also nette app) there, so this is what I want, I think. Because not only latte/latte is important, but also nette/forms, nette/application and other packages which can register macros / extensions / providers.

Just maybe consider if we shouldn't do this only once, not for all templates compiled.

Reply:
Good point. We shoukd xompute cache key in constructor from static things and just combine it with template specific things in compile